More about authenticating Version Negotiation (#3871)
quicwg/base-drafts: draft-ietf-quic-invariants.md

Supported Version fields, or a truncated Supported Version.

Version Negotiation packets do not use integrity or confidentiality protection.
Specific QUIC versions might include protocol elements that allow endpoints to

> detect when the set of supported versions is modified or corrupted.

Reviewer: ```suggestion
detect modification or corruption in the set of supported versions.
```

The Version Negotiation packet described in this document is not
integrity-protected; it only has modest protection against insertion by off-path

> attackers.  A QUIC version that uses a Version Negotiation packet MUST define a

Reviewer 2: One could still argue that QUIC v1 "uses" the VN packet to indicate lack of support.

Reviewer 2: ```suggestion
attackers.  A specification which permits client to change the QUIC version in use
based on the contents of a Version Negotiation packet MUST also define a
```
Something like this?

> mechanism to verify authenticate the values the packet contains.

Author: ```suggestion
mechanism to authenticate the values the packet contains.
```
